graylog2-web-interface/src/components/datanode/DataNodeList/DataNodeActions.tsx

```diff
@@ -140,16 +159,42 @@ const DataNodeActions = ({ dataNode, displayAs }: Props) => {
           {isDatanodeRunning && <ActionButton onClick={() => handleAction(DIALOG_TYPES.STOP)} bsSize="small">Stop</ActionButton>}
           {isDatanodeRemoved && <ActionButton onClick={() => handleAction(DIALOG_TYPES.REJOIN)} bsSize="small">Rejoin</ActionButton>}
           {(!isDatanodeRemoved || isRemovingDatanode) && <ActionButton onClick={() => handleAction(DIALOG_TYPES.REMOVE)} bsSize="small">Remove</ActionButton>}
+          <ActionButton onClick={() => setShowLogsDialog(true)} bsSize="small">Show logs</ActionButton>
         </>
       )}
-      {showDialog && (
+      {showConfirmDialog && (
         <ConfirmDialog title={DIALOG_TEXT[dialogType].dialogTitle}
                        show
                        onConfirm={handleConfirm}
                        onCancel={handleClearState}>
           {DIALOG_TEXT[dialogType].dialogBody(dataNode.hostname)}
         </ConfirmDialog>
       )}
+      {showLogsDialog && (
+        <BootstrapModalWrapper showModal={showLogsDialog}
+                               onHide={() => setShowLogsDialog(false)}
+                               bsSize="large"
+                               backdrop>
+          <Modal.Header closeButton>
+            <Modal.Title>{dataNode.hostname} logs</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <ButtonGroup>
+              <Button active={logsType === 'stdout'} onClick={() => setLogsType('stdout')}>STDOUT</Button>
+              <Button active={logsType === 'stderr'} onClick={() => setLogsType('stderr')}>STDERR</Button>
+            </ButtonGroup>
+            <pre>
+              {logs[logsType] ? (
```

Comment:
I think we should extract this section to `DataNodeLogsView` and use the hook in there so that the Actions component is cleaner.